src/content/platform-integration/web/embedding-flutter-web.md

```diff
@@ -0,0 +1,326 @@
+
+### Enable multi-view mode
+
+Enable multi-view mode setting `multiViewEnabled: true` in the
+`initializeEngine` method as shown:
+
+```js highlightLines=8
+// flutter_bootstrap.js
+{% raw %}{{flutter_js}}{% endraw %}
+{% raw %}{{flutter_build_config}}{% endraw %}
+
+_flutter.loader.load({
+  onEntrypointLoaded: async function onEntrypointLoaded(engineInitializer) {
+    let engine = await engineInitializer.initializeEngine({
+      multiViewEnabled: true, // Enables embedded mode.
+    });
+    let app = await engine.runApp();
+    // Make this `app` object available to your JS app.
+  }
+});
+```
+
+### Manage Flutter views from JS
+
+To add or remove views, use the `app` object returned by the `runApp` method:
+
+```js highlightLines=2-4,7
+// Adding a view...
+let viewId = app.addView({
+  hostElement: document.querySelector('#some-element'),
+});
+
+// Removing viewId...
+let viewConfig = flutterApp.removeView(viewId);
+```
+
+### Handling view changes from Dart
+
+View additions and removals are surfaced to Flutter through the
+[`didChangeMetrics` method][] of the `WidgetsBinding` class.
+
+The complete list of views attached to your Flutter app is available
+through the `WidgetsBinding.instance.platformDispatcher.views` iterable. These
+views are of [type `FlutterView`][].
+
+To render content into each `FlutterView`, your Flutter app needs to create a
+[`View` widget][]. `View` widgets can be grouped together under a
+[`ViewCollection` widget][].
+
+The following is an example from the _Multi View Playground_ that encapsulates
+the above in a `MultiViewApp` widget that can be used as the root widget for
+your app. A [`WidgetBuilder` function][] will run for each `FlutterView`:
+
+```dart highlightLines=25,39,46-49,56-61,72
+// multi_view_app.dart
+
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show FlutterView;
+import 'package:flutter/widgets.dart';
+
+/// Calls [viewBuilder] for every view added to the app to obtain the widget to
+/// render into that view. The current view can be looked up with [View.of].
+class MultiViewApp extends StatefulWidget {
+  const MultiViewApp({super.key, required this.viewBuilder});
+
+  final WidgetBuilder viewBuilder;
+
+  @override
+  State<MultiViewApp> createState() => _MultiViewAppState();
+}
+
+class _MultiViewAppState extends State<MultiViewApp> with WidgetsBindingObserver {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+    _updateViews();
+  }
+
+  @override
+  void didUpdateWidget(MultiViewApp oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    // Need to re-evaluate the viewBuilder callback for all views.
+    _views.clear();
+    _updateViews();
+  }
+
+  @override
+  void didChangeMetrics() {
+    _updateViews();
+  }
+
+  Map<Object, Widget> _views = <Object, Widget>{};
```

Comment:
```suggestion
  final Map<Object, Widget> _views = <Object, Widget>{};
```

Reply:
Can't make this one `final`, it's re-assigned in the `setState()` call of `_updateViews()`.